### PR TITLE
Allow to import unarmored S/MIME certificate

### LINKED ISSUE #3559: [priority] s/mime cert could not be imported - debug
I've sent details and sample cert to your FlowCrypt email now. Thank you!

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR allows to import a certificate in DER format
in addition to already implemented PEM and PKCS#12 PFX formats

issue #3559

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by rrrooommmaaa** [comment]
I'll also add advanced email extraction to completely support the provided certificate.
The e-mail there isn't stored in "Subject", rather in "Subject Alternative Name" in the format
"RFC822 Name=..."

**Comment by tomholub** [comment]
Excellent!

File: extension/js/common/core/crypto/smime/smime-key.ts [hunk 1]
```diff
@@ -84,14 +91,26 @@ export class SmimeKey {
...
     }
+    throw new UnreportableError(`This S/MIME x.509 certificate has an invalid recipient email: ${emailFromSubject}`);
+  }
+
+  private static getKeyFromCertificate = (certificate: forge.pki.Certificate, pem: string): Key => {
+    SmimeKey.removeWeakKeys(certificate);
+    const emails = SmimeKey.getNormalizedEmailsFromCertificate(certificate);
     const key = {
       type: 'x509',
       id: certificate.serialNumber.toUpperCase(),
```
Comment (rrrooommmaaa) [inline extension/js/common/core/crypto/smime/smime-key.ts:116]: I'm afraid we can't use `serialNumber` as `fingerprint`.
Should we open a new task for this as well as include migration to re-calculate ids?

**Comment by rrrooommmaaa** [comment]
We can merge this, but the original issue is still unresolved due to `serialNumber` vulnerability

File: extension/js/common/core/crypto/smime/smime-key.ts (hunk 1 above)
Comment (tomholub) [reply extension/js/common/core/crypto/smime/smime-key.ts:116]: Created an issue to discuss

**Review by tomholub — APPROVED** [review]
Thanks!

**Pull request merged**